jme3-effects/src/main/java/com/jme3/post/filters/ContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,309 @@
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.post.filters;
+
+import com.jme3.asset.AssetManager;
+import com.jme3.export.InputCapsule;
+import com.jme3.export.JmeExporter;
+import com.jme3.export.JmeImporter;
+import com.jme3.export.OutputCapsule;
+import com.jme3.material.Material;
+import com.jme3.math.Vector2f;
+import com.jme3.math.Vector3f;
+import com.jme3.post.Filter;
+import com.jme3.renderer.RenderManager;
+import com.jme3.renderer.ViewPort;
+import java.io.IOException;
+
+/**
+ * A color filter used to change the contrast for each color channel independently and the brightness of colors
+ * of the scene textures based on a simple transfer function.
+ *
+ * The transfer function involves setting the brightness of colors based on a maxValue and a minValue,
+ * then setting the contrast of each color channel independently using a power law function before the textures' colors are processed by the
+ * rasterizer and the final result can be scaled again on runtime using scale factors from 0.0 to 1.0.
+ *
+ * @author pavl_g.
+ */
+public class ContrastAdjustmentFilter extends Filter {
+    //the different channels exponents
+    protected float exp_r = 2.2f;
+    protected float exp_g = 2.2f;
+    protected float exp_b = 2.2f;
+    //the minimum value and the maximum value of the brightness parameter
+    protected float minBrightness = 0f;
+    protected float maxBrightness = 1f;
+    //the final pass scale factor
+    protected float scale_r = 1f;
+    protected float scale_b = 1f;
+    protected float scale_g = 1f;
+    protected Material material;
+
+    /**
+     * Instantiates a default color contrast filter, default brightness and default scale.
+     * Default values :
+     * - Exponents = 2.2 on all channels.
+     * - Brightness = 1.0f (normal).
+     * - Scale = 1.0f.
+     */
+    public ContrastAdjustmentFilter() {
+        super(ContrastAdjustmentFilter.class.getName());
+    }
+
+    /**
+     * Instantiates a color contrast filter with a default brightness and default scale.
+     * Default values :
+     * - Brightness = 1.0f (normal).
+     * - Scale = 1.0f.
+     *
+     * @param exp_r the red color exponent.
+     * @param exp_b the blue color exponent.
+     * @param exp_g the green color exponent.
+     */
+    public ContrastAdjustmentFilter(float exp_r, float exp_b, float exp_g) {
+        super(ContrastAdjustmentFilter.class.getName());
+        this.exp_r = exp_r;
+        this.exp_g = exp_g;
+        this.exp_b = exp_b;
+    }
+
+    /**
+     * Instantiates a color contrast filter by adjusting different parameters.
+     *
+     * @param exponents  the exponents applied to the colors in order, where x = r, y = g, z = b.
+     * @param brightness the brightness applied to the textures in order, where x = minBrightness, y = maxBrightness.
```

Comment:
`brightness` is not applied to textures, but to a rendered scene.

"in order" adds no useful information, since max and min do not have a natural ordering.

Also, should clarify that `brightness` applies to the input levels, not output levels.